lib/nimbase.h

```diff
@@ -547,7 +547,7 @@ static inline void GCGuard (void *ptr) { asm volatile ("" :: "X" (ptr)); }
 #endif
 
 // Test to see if Nim and the C compiler agree on the size of a pointer.
-NIM_STATIC_ASSERT(sizeof(NI) == sizeof(void*) && NIM_INTBITS == sizeof(NI)*8, "");
+NIM_STATIC_ASSERT(sizeof(NI) == sizeof(void*) && NIM_INTBITS == sizeof(NI)*8, "Pointer size mismatch between Nim and C/C++ backend. You probably need to setup the backend compiler for target CPU.");
```

Comment:
Is there a way to provide a more informative message, such as "Pointer size mismatch [...] . The Nim compiler targets <arch>, while the C compiler targets <arch>"?

Reply:
Getting target CPU or OS of currently using backend C compiler would be hard.
I think it would require compiler specific code.

For example:

```c
#ifdef __GNUC__
#  ifdef __x86_64
#    define C_BACKEND_TARGET_CPU "x86_64"
#  elif defined __aarch64__
#    define C_BACKEND_TARGET_CPU "arm64"
#  endif
#elif defined ...
#endif
```

(If gcc or clang is installed, `cpp -dM test.c` prints all predefined macros. They can be used to detect target CPU or OS.)

Or parsing output of `gcc -v` or `clang -v`.

And this assert just checks pointer size. Doesn't check whether Nim's target CPU and backend C compiler target CPU match.
I can run `nim c -r --cpu=arm64 test.nim` on x86_64 CPU and I get an executable for x86_64 CPU without error.
I can check target CPU of an executable file with `file myprog` or `objdump -p myprog`.

Do Nim really should check whether target CPU/OS of Nim and backend C compiler matches even if it require large code?